src/vm/tests/simple_apply_eval.rs

```diff
@@ -929,6 +929,36 @@ fn test_lets() {
         .for_each(|(program, expectation)| assert_eq!(expectation.clone(), execute(program)));
 }
 
+#[test]
+fn test_2239_merge() {
+    let tests = [
+        "(define-data-var a {p: uint} (merge {p: 2} {p: u2})) (var-get a)",
+        "(merge {p: 2} {p: u2})",
+        "(merge {p: 2} {q: 3})",
+        "(define-data-var c {p: uint} {p: u2}) (var-get c)",
+        "(define-data-var d {p: uint} (merge {p: u2} {p: u2})) (var-get d)",
+        "(define-data-var e {p: int, q: int} {p: 2, q: 3}) (var-get e)",
+        "(define-data-var f {p: int, q: int} (merge {q: 2, p: 3} {p: 4})) (var-get f)",
```

Comment:
Do you maybe want to add the specific test that triggered this problem?

Reply:
It's the first test in the list.